md-dao/src/main/java/com/wipro/ats/bdre/md/dao/ProcessDAO.java

```diff
@@ -702,5 +729,155 @@ else  if (childProcess.getProcessType().getProcessTypeId() == 36){
+            }
+            case "execute": if (executeList.contains(process.getPermissionTypeByUserAccessId().getPermissionTypeId())||
+                    (userRolesNameList.contains(processCreater)&&executeList.contains(process.getPermissionTypeByGroupAccessId().getPermissionTypeId()))
+                    || (!userRolesNameList.contains(processCreater)&&executeList.contains(process.getPermissionTypeByOthersAccessId().getPermissionTypeId()))
+                    )
+            {
+                return ACCESSGRANTED;
+            }
+            else
+            {
+                LOGGER.info("user execute");
+                throw new SecurityException(ACCESSDENIED);
+            }
+            default: {LOGGER.info(" no operation");
+                throw new SecurityException("no  operation");
+            }
+
+        }
+
+
+    }
+    else{
+    if (userRolesNameList.contains(processCreater))
+    {
+        switch (action){
+            case "write": if (writeList.contains(process.getPermissionTypeByGroupAccessId().getPermissionTypeId()))
+                               {
+                                   return ACCESSGRANTED;
+                               }
+                           else
+                          {
+                              LOGGER.info("group write");
+                              throw new SecurityException(ACCESSDENIED);
+                          }
+            case "read": if (readList.contains(process.getPermissionTypeByGroupAccessId().getPermissionTypeId()))
+                               {
+                                   return ACCESSGRANTED;
+                               }
+                           else
+                          {
+                              LOGGER.info("group read");
+                              throw new SecurityException(ACCESSDENIED);
+                          }
+            case "execute": if (executeList.contains(process.getPermissionTypeByGroupAccessId().getPermissionTypeId()))
+                            {
+                                return ACCESSGRANTED;
+                            }
+                            else
+                            {
+                                LOGGER.info("group execute");
+                                throw new SecurityException(ACCESSDENIED);
+                            }
+            default: {LOGGER.info("no   operation");
+                throw new SecurityException(" no operation");
+            }
+        }
+    }
+    else {
+        switch (action){
+            case "write": if (writeList.contains(process.getPermissionTypeByOthersAccessId().getPermissionTypeId()))
+            {
+                return ACCESSGRANTED;
+            }
+            else
+            {
+                LOGGER.info("other write");
+                throw new SecurityException(ACCESSDENIED);
+            }
+            case "read": if (readList.contains(process.getPermissionTypeByOthersAccessId().getPermissionTypeId()))
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 9 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1151)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)